ibis/expr/datatypes.py

```diff
@@ -1,47 +1,72 @@
+import builtins
 import collections
 import datetime
+import functools
 import itertools
 import numbers
 import re
+import typing
+
+from typing import (
+    Any as GenericAny,
+    Callable,
+    Iterator,
+    List,
+    Mapping,
+    NamedTuple,
+    Optional,
+    Sequence,
+    Set as GenericSet,
+    Tuple,
+    TypeVar,
+    Union,
+)
 
-import six
 import pandas as pd
 
 import toolz
 from multipledispatch import Dispatcher
 
 import ibis.common as com
-from ibis.compat import PY2, builtins, functools
 import ibis.expr.types as ir
 
 
-class DataType(object):
-
+class DataType:
     __slots__ = 'nullable',
 
-    def __init__(self, nullable=True):
+    def __init__(self, nullable: bool = True) -> None:
         self.nullable = nullable
 
-    def __call__(self, nullable=True):
+    def __call__(self, nullable: bool = True) -> 'DataType':
+        if nullable is not True and nullable is not False:
```

Comment:
Is it faster than `type(x) == bool`?

Reply:
Probably. Two pointer comparisons vs a function call + a pointer comparison.

Reply:
Fair enough.